redfish-core/lib/fabric_adapters.hpp

```diff
@@ -199,6 +199,77 @@ inline void linkAsPCIeDevice(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
         deviceArray.size();
 }
 
+inline void getFabricAdapterPCIeSlots(
+    const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+    const std::string& systemName, const std::string& fabricAdapterPath,
+    std::function<void(const std::string& /*chassisName*/,
+                       const dbus::utility::MapperEndPoints&)>&& callback)
+{
+    constexpr std::array<std::string_view, 1> pcieSlotInterface{
+        "xyz.openbmc_project.Inventory.Item.PCIeSlot"};
+    dbus::utility::getAssociatedSubTreePaths(
+        fabricAdapterPath + "/containing",
+        sdbusplus::message::object_path("/xyz/openbmc_project/inventory"), 0,
+        pcieSlotInterface,
+        [aResp, systemName, fabricAdapterPath, callback{std::move(callback)}](
+            const boost::system::error_code& ec,
+            const dbus::utility::MapperEndPoints& pcieSlotPaths) {
+        if (ec)
+        {
+            if (ec.value() == EBADR)
+            {
+                BMCWEB_LOG_DEBUG << "Slot association not found";
+                return;
+            }
+            BMCWEB_LOG_ERROR << "DBUS response error " << ec.value();
+            messages::internalError(aResp->res);
+            return;
+        }
+        if (pcieSlotPaths.empty())
+        {
+            // no slot associations
+            BMCWEB_LOG_DEBUG << "Slot association not found";
+            return;
+        }
+
+        // There are PCIeSlots.
+        // Link it to PCIeSlot of the associated chassis
```

Comment:
The code does not perform as described in the comments. It is querying the chassis associated with fabric adapters. It appears to me that both getAssociatedSubTreePaths calls are unrelated.

Reply:
PCIeSlots are associated with "/containing" under chassis.
This function is to get PCIeSlots under chassis, as the comment indicates.